Core/Utilities/JEncode.py

```diff
@@ -184,3 +184,18 @@ def decode(encodedData):
       but it is for compatibility
   """
   return json.loads(encodedData, cls=DJSONDecoder), len(encodedData)
+
+
+def strToIntDict(inDict):
+  """ Because JSON will transform dict with int keys to str keys,
+      this utility method is just to cast it back.
+      This shows useful in cases when sending dict indexed on
+      jobID or requestID for example
+
+      :param inDict: dictionnary with strings as keys e.g. {'1': 1, '2': 2}
+
+      :returns: dictionnary with int as keys e.g. {1: 1, 2: 2}
+
+  """
+
+  return dict((int(key), value) for key, value in inDict.iteritems())
```

Comment:
Change to dictionary comprehension and does it need to use `iteritems`?
```suggestion
  return {int(key): value for key, value in inDict.items()}
```

Reply:
OK for the dict comprehension, but using `items` makes an extra copy which seems useless to me. Unless I misunderstand something ?

Reply:
The reason for suggesting `items` is that `iteritems` doesn't exist in Python 3 (well `items` is `iteritems`). It also doesn't make much of a difference as copying these objects is very cheap (especially relative to changing it to a dictionary comprehension):
```python
> import numpy as np
> data = dict(np.random.randint(10000, size=(100, 2)))
> %timeit dict((int(key), value) for key, value in data.items())
10000 loops, best of 3: 42.6 µs per loop
> %timeit dict((int(key), value) for key, value in data.iteritems())
10000 loops, best of 3: 42.4 µs per loop
> %timeit {int(key): value for key, value in data.items()}
10000 loops, best of 3: 34.8 µs per loop
> %timeit {int(key): value for key, value in data.iteritems()}
10000 loops, best of 3: 33.5 µs per loop
```

Reply:
yes, but the reason for doing so was (besides good practice) that we wanted to use `2to3`, and `2to3` works better when there is a proper usage of `items` vs `iteritems` (https://docs.python.org/2/library/2to3.html#2to3fixer-dict)